### PR TITLE
invites: refactor to reflect accurate status 

### PR DESCRIPTION
**Pull request by arthyn**
This PR attempts to fix both #940 and #947. Any logic dealing with invites was moved to the useInvites hook to help get useRoller smaller and clean up some of the component files. 

Invites now only ever get added if they are correct, so we either show a status of 'updating' or we show the correct number. Also when generating invites for the first time, we give a modal explaining what's happening and showing progress.

In addition, I did a bit of drive by refactoring to convert some files to Typescript and also removing any unused files, like the old useInvites and anything to do with L1 invites.

### CODE REVIEW

**Comment by arthyn** [comment]
Also fixes #978

File: src/views/Invite/GenerateInvites.tsx [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+                  }}
+                />
+                planet invite code{numInvites > 1 ? 's' : ''}
+              </Row>
+            ) : (
+              <Box className="migration-prompt">
+                ETH fees are very high currently. You can spawn invites for free
+                after{' '}
+                <span className="migrate" onClick={() => null}>
+                  {/* TODO: change this to navigate to migration */}
```
Comment (tomholford) [inline src/views/Invite/GenerateInvites.tsx:102]: Is this still needed?

File: src/views/Invite/useInvites.ts [hunk 2]
```diff
@@ -0,0 +1,372 @@
...
+    });
+  },
+  setInvites: (point: number, invites: Invite[]) =>
+    set(state => {
+      const newInvites: Invites = {};
+      newInvites[point] = invites;
+      return { invites: Object.assign(state.invites, newInvites) };
+    }),
+  updateJob: (point: number, updates: Partial<InviteJob>) => {
+    debugger;
```
Comment (tomholford) [inline src/views/Invite/useInvites.ts:84]:
```suggestion
```

File: src/Bridge.tsx [hunk 3]
```diff
@@ -103,16 +105,24 @@ function Bridge() {
...
-          <Box width="280px">{modalText}</Box>
-        </Modal>
-        <LoadingOverlay loading={loading} />
-        <Poller />
+        <BrowserRouter>
+          <IndigoApp>
+            <ThemeProvider theme={light}>
+              <Router />
+              <Routes>
+                <Route path="/" element={null} />
```
Comment (tomholford) [inline src/Bridge.tsx:113]: Do we need a fallback for the root route? 

Something like this:
```tsx
  <Route path="*" element={<Navigate to={'/'} />} />
```

**Review by tomholford — APPROVED** [review]
Overall, LGTM. A few questions inline. Happy to see all these much-needed changes. The TS and Router will make Bridge much more maintainable going forward.

Pulled the branch and tested with `npm run pilot-ropsten`, the invite flow is much smoother. Nice work @arthyn :muscle:

File: src/views/Invite/GenerateInvites.tsx (hunk 1 above)
Comment (arthyn) [reply src/views/Invite/GenerateInvites.tsx:102]: I forgot to rip it out after our discussion in the bridge channel. I'll do that and re-request a review.

**Review by tomholford — APPROVED** [review]
Pulled and tested invite creation flow on Ropsten, changes LGTM. Nice work @arthyn :raised_hands:

**Pull request merged**